straxen/plugins/event_processing.py

```diff
@@ -222,103 +248,159 @@ def infer_dtype(self):
+                                                  field_names,
+                                                  peak_properties_to_save)
+
+    @staticmethod
+    @numba.njit
+    def set_event_properties(result, largest_s1s, largest_s2s, peaks):
+        """Get properties like drift time and area before main S2"""
         # Compute drift times only if we have a valid S1-S2 pair
-        if len(main_s) == 2:
-            result['drift_time'] = \
-                main_s[2]['center_time'] - main_s[1]['center_time']
-            if 1 in secondary_s:
-                result['alt_s1_interaction_drift_time'] = \
-                    main_s[2]['center_time'] - secondary_s[1]['center_time']
-            if 2 in secondary_s:
-                result['alt_s2_interaction_drift_time'] = \
-                    secondary_s[2]['center_time'] - main_s[1]['center_time']
+        if len(largest_s1s) > 0 and len(largest_s2s) > 0:
+            result['drift_time'] = largest_s2s[0]['center_time'] - largest_s1s[0]['center_time']
+            if len(largest_s1s) > 1:
+                result['alt_s1_interaction_drift_time'] = largest_s2s[0]['center_time'] - largest_s1s[1]['center_time']
+            if len(largest_s2s) > 1:
+                result['alt_s2_interaction_drift_time'] = largest_s2s[1]['center_time'] - largest_s1s[0]['center_time']
 
         # areas before main S2
-        if result[f's2_index'] != -1:
-            peaks_before_ms2 = peaks[peaks['time'] < main_s[2]['time']]
-            result['area_before_main_s2'] = sum(peaks_before_ms2['area'])
+        if len(largest_s2s):
+            peaks_before_ms2 = peaks[peaks['time'] < largest_s2s[0]['time']]
+            result['area_before_main_s2'] = np.sum(peaks_before_ms2['area'])
 
             s2peaks_before_ms2 = peaks_before_ms2[peaks_before_ms2['type'] == 2]
             if len(s2peaks_before_ms2) == 0:
                 result['large_s2_before_main_s2'] = 0
             else:
-                result['large_s2_before_main_s2'] = max(s2peaks_before_ms2['area'])
-
+                result['large_s2_before_main_s2'] = np.max(s2peaks_before_ms2['area'])
         return result
 
+    @staticmethod
+    # @numba.njit <- works but slows if fill_events is not numbafied
+    def get_largest_sx_peaks(peaks,
+                             s_i,
+                             s1_before_time=np.inf,
+                             s1_min_coincidence=0,
+                             number_of_peaks=2):
+        """Get the largest S1/S2. For S1s allow a min coincidence and max time"""
+        # Find all peaks of this type (S1 or S2)
+        s_mask = peaks['type'] == s_i
+        if s_i == 1:
+            s_mask &= peaks['time'] < s1_before_time
+            s_mask &= peaks['tight_coincidence'] >= s1_min_coincidence
+
+        selected_peaks = peaks[s_mask]
+        s_index = np.arange(len(peaks))[s_mask]
+        largest_peaks = np.argsort(selected_peaks['area'])[-number_of_peaks:][::-1]
```

Comment:
```suggestion
        largest_peaks = np.argsort(selected_peaks['area'])[::-1][:number_of_peaks]
```
Does not this work too? Minor change but, the other way around it took me a moment to understand :D

Reply:
Right thats a good suggestion!